src/v/cluster/jumbo_log_frontend.cc

```diff
@@ -0,0 +1,215 @@
+// Copyright 2020 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "cluster/jumbo_log_frontend.h"
+
+#include "base/outcome.h"
+#include "base/vassert.h"
+#include "base/vlog.h"
+#include "cluster/controller.h"
+#include "cluster/jumbo_log_service.h"
+#include "cluster/jumbo_log_stm.h"
+#include "cluster/logger.h"
+#include "cluster/metadata_cache.h"
+#include "cluster/partition_leaders_table.h"
+#include "cluster/partition_manager.h"
+#include "cluster/shard_table.h"
+#include "cluster/topics_frontend.h"
+#include "cluster/types.h"
+#include "config/configuration.h"
+#include "errc.h"
+#include "jl/metadata.h"
+#include "jl/rpc.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "model/namespace.h"
+#include "model/timeout_clock.h"
+#include "rpc/connection_cache.h"
+#include "rpc/types.h"
+#include "ssx/future-util.h"
+#include "topic_configuration.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/shard_id.hh>
+#include <seastar/core/sharded.hh>
+#include <seastar/core/smp.hh>
+
+#include <__utility/move.h>
+
+#include <chrono>
+#include <exception>
+#include <memory>
+#include <system_error>
+#include <tuple>
+#include <type_traits>
+#include <vector>
+
+namespace cluster {
+using namespace std::chrono_literals;
+
+cluster::errc map_errc_fixme(std::error_code ec);
+
+ss::future<
+  result<rpc::client_context<jumbo_log::rpc::create_write_intent_reply>>>
+create_write_intent_handler::dispatch(
+  jumbo_log::rpc::jumbo_log_client_protocol proto,
+  jumbo_log::rpc::create_write_intent_request req,
+  model::timeout_clock::duration timeout) {
+    req.timeout = timeout;
+    return proto.create_write_intent(
+      std::move(req), rpc::client_opts(model::timeout_clock::now() + timeout));
+}
+
+static_assert(
+  std::is_trivially_copyable<jumbo_log::rpc::create_write_intent_request>(),
+  "The code bellow assumes that create_write_intent_request is trivially "
+  "copyable. If this is not the case, the code should be updated to use "
+  "std::move semantics.");
```

Comment:
I'd love some advice on this. Currently my IDE draws squiggly lines below std::move invocation saying that it is a trivially copiable type so std::move shouldn't be used. I usually listen to the IDE so I remove std::move invocations. But if I do that and the type changes to not be a trivially copyable type the code wouldn't be as efficient as it could have been, right? This static_assert is meant to catch that and prompt the author to add std::move back where needed.

Thoughts?